lib/rubocop/cop/style/class_structure.rb

```diff
@@ -0,0 +1,312 @@
+
+        def ignore?(classification)
+          classification.nil? ||
+            classification.to_s.end_with?('=') ||
+            expected_order.index(classification).nil?
+        end
+
+        def node_visibility(node)
+          _, method_name, = *find_visibility_start(node)
+          method_name || :public
+        end
+
+        def find_visibility_start(node)
+          left_siblings_of(node).reverse.find do |parent_node|
+            if parent_node.send_type?
+              _, method_name, = *parent_node
+              %i[private protected public].include?(method_name)
+            end
+          end
+        end
+
+        # Navigate to find the last protected method
+        def find_visibility_end(node)
+          possible_visibilities = VISIBILITIES - [node_visibility(node)]
+          right = right_siblings_of(node)
+          right.find do |child_node|
+            if child_node.def_type?
+              possible_visibilities.include?(node_visibility(child_node))
+            end
+          end || right.last
+        end
+
+        def siblings_of(node)
+          node.parent.children
+        end
+
+        def right_siblings_of(node)
+          siblings_of(node)[node.sibling_index..-1]
+        end
+
+        def left_siblings_of(node)
+          siblings_of(node)[0, node.sibling_index]
+        end
+
+        def humanize_node(node)
+          method_name, = *node
+          if node.def_type?
+            return :initialize if method_name == :initialize
+            return "#{node_visibility(node)}_method"
+          end
+          HUMANIZED_NODE_TYPE[node.type] || node.type
+        end
+
+        def source_range_with_comment(node)
+          begin_pos, end_pos =
+            if node.def_type?
+              start_node = find_visibility_start(node) || node
+              end_node = find_visibility_end(node) || node
+              [begin_pos_with_comment(start_node),
+               end_position_for(end_node) + 1]
+            else
+              [begin_pos_with_comment(node), end_position_for(node)]
+            end
+
+          Parser::Source::Range.new(buffer, begin_pos, end_pos)
+        end
+
+        def end_position_for(node)
+          end_line = buffer.line_for_position(node.loc.expression.end_pos)
+          buffer.line_range(end_line).end_pos
+        end
+
+        def begin_pos_with_comment(node)
+          annotation_line = node.loc.line - 1
+          first_comment = nil
+
+          comments_before_line(annotation_line).reverse_each do |comment|
+            if comment.location.line == annotation_line
+              first_comment = comment
+              annotation_line -= 1
+            end
+          end
+
+          start_line_position(first_comment || node)
+        end
+
+        def start_line_position(node)
+          buffer.line_range(node.loc.line).begin_pos - 1
+        end
+
+        def comments_before_line(line)
+          processed_source.comments.lazy.select { |c| c.location.line <= line }
```

Comment:
I don't think the addition of `#lazy` accomplishes much here. Hopefully the number of comments in the source file is finite. 😅